src/util/InputRangeUtils.h

```diff
@@ -225,4 +229,36 @@ template <typename Range, typename F>
 CachingContinuableTransformInputRange(Range&&, F)
     -> CachingContinuableTransformInputRange<all_t<Range>, F>;
 
+// This class can be used to concatenate input ranges into a single range.
+// Elements within ranges are iterated in order until their end, at which point
+// the next element will be the first element of the next range. When all
+// elements of all ranges have been iterated, nullopt is returned for all
+// subsequent get() calls. Note: This class only works with Views that implement
+// "get()" like the InputRangeFromGet subclasses.
+CPP_class_template(typename ViewType)(
+    requires(ql::ranges::input_range<ViewType>)) struct ConcatenatedInputRange
+    : ad_utility::InputRangeFromGet<ql::ranges::range_value_t<ViewType>> {
+  using ViewCollection = std::vector<std::unique_ptr<ViewType>>;
+  ViewCollection viewCollection_;
+  ql::ranges::iterator_t<ViewCollection> collectionIt_;
+
+  ConcatenatedInputRange(ViewCollection viewCollection)
+      : viewCollection_(std::move(viewCollection)),
+        collectionIt_(viewCollection_.begin()) {}
+
+  std::optional<ql::ranges::range_value_t<ViewType>> get() {
+    while (collectionIt_ != viewCollection_.end()) {
+      auto elementOpt = (*collectionIt_)->get();
+      if (elementOpt.has_value()) {
+        // Element exists in the currently iterated view
+        return elementOpt;
+      }
+      // Done with current view, but not all views
+      ++collectionIt_;
+    }
+    // Done iterating over all views
+    return std::nullopt;
+  }
+};
```

Comment:
What I just found out....

Can't we just use a `LazyResult{ql::views::join(vectorOfLazyResults....))`

I am sorry for the confusion. `Concat` typically implies
heterogenous ranges (in which case we need a more complicated implementation).
Sorry for the confusion, maybe we can get rid of this abstraction after all and simply use `views::join(vector<TypeErased>)` for the complex cases where the overhead is neglectable.